content/articles/Understanding-React-UseEffect/index.md

```diff
@@ -0,0 +1,554 @@
+### Using React Hooks: Understanding React UseEffect and creating custom Hooks
+
+---
+layout: engineering-education
+status: publish
+published: true
+url: /understanding-react-useeffect/
+title: Understanding React UseEffect and creating custom Hooks
+description: In this article we will look at what hooks are, how they work and their benefits. We will also demonstrate how useEffect can manage side effects from components.
+author: zuko-obi
+date: 2021-11-16T00:00:00-12:03
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/understanding-react-useeffect/hero.jpg
+    alt: Understanding React UseEffect and creating custom Hooks Hero Image
+---
+React Hooks are special functions that allow you to "hook into" React features without the need to create classes. They provide an alternative to writing class-based components (as hooks do not function inside classes).
+<!--more-->
+Operations like data fetching or manually changing the Document Object model (DOM) via React components can cause side effects or "effects" as they are known. This can affect other components within your application.
+
+React comes with in-built hooks like `useEffect` that allow you to perform side effects from a function or component. With a single effect function, we can achieve the purposes of the React classes. This includes `componentDidMount`, `componentWillUnmount`, and `componentDidUpdate`.
+
+
+### Key takeaways
+By the end of this tutorial, the reader will be able to understand:
+- What React hooks are and their benefits.
+- The functionalities of `useEffect`.
+- Rules for using `useEffect`.
+- How to consume APIs with `useEffect` and process responses.
+- How to use `useEffect` to "clean up" effects (or "after effects" as I call them) by returning a function.
+
+### Prerequisites
+To follow through with this article, the reader should have:
+- React v16.0 or newer installed.
+- An understanding of Javascript.
+- A Basic knowledge of ReactJS.
+- A suitable text editor of your choice.
+
+### What are React hooks?
+React Hooks are the best thing to happen to React devs in a long time. It makes it possible for functional programmers to create dynamic projects. This happens without having to write classes as class-based components need a `render() method`, have complex UI logic and are generally more complex to manage.
+
+You create a hook depending on its utility.
+
+### What is the React UseEffect?
+The useEffect hook has superpowers that enable us to design our custom hooks. When a change occurs, it allows us to perform side effects in functional components. It allows data retrieval, DOM modification, and function execution each time a component renders.
+
+Let us explore some examples:
+
+```javascript
+
+import React, { useState, useEffect } from &quot;react&quot;;
```

Comment:
Kindly use the normal quotation make ` "example" `  instead of `&quot`. Markdown will interprate the signs appropriately. Make this change throughout your whole code in the tutorial.

Reply:
done/